probe-rs/src/probe/jlink/mod.rs

```diff
@@ -465,19 +503,39 @@ impl DebugProbe for JLink {
     }
 
     fn get_interface_dap(&self) -> Option<&dyn DAPAccess> {
-        Some(self as _)
+        // For now, we only support using SWD for ARM chips, but
+        // JTAG would be possible as well.
+        if self.supported_protocols.contains(&WireProtocol::Swd) {
```

Comment:
I like this change for now, since it's preventing missuse.

Theoretically it is possible to use DAP over JTag, afaik, no? I wonder if we ever need this (does anyone do JTag with ARM cores?)

Reply:
DAP over JTAG is possible, but I have no idea how many people use it.